cue/schemas/panels/bar/migrate/migrate.cue

```diff
@@ -24,8 +24,8 @@ kind: "BarChart"
 spec: {
 	calculation: *commonMigrate.#mapping.calc[#panel.options.reduceOptions.calcs[0]] | commonMigrate.#defaultCalc // only consider [0] here as Perses's GaugeChart doesn't support individual calcs
 
-	#unit: *commonMigrate.#mapping.unit[#panel.fieldConfig.defaults.unit] | null
-	if #unit != null {
+	#unit: *commonMigrate.#mapping.unit[#panel.fieldConfig.defaults.unit] | "decimal"
```

Comment:
this fix works however it would be better to keep not creating a format when there's no unit nor decimal

Thus please consider this fix instead:
```suggestion
	#unit: *commonMigrate.#mapping.unit[#panel.fieldConfig.defaults.unit] | null
	if #unit != null {
		format: unit: #unit
	}

	#decimal: *#panel.fieldConfig.defaults.decimal | null
	if #decimal != null {
		format: {
			decimalPlaces: #decimal
			if #unit == null {
				unit: "decimal"
			}
		}
	}
```

Reply:
Yes I see which case you faced, basically `format` is an optional attribute but if ever it is present, a value for `unit` has to be provided. My proposed change also fixes your issue, but enforces this "decimal" default value only when there's something to map to decimalPlaces - instead of systematically with your proposed logic.